include/RED4ext/GpuApi/CommandListContext.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <RED4ext/DynArray.hpp>
+#include <REd4ext/CString.hpp>
+#include <d3d12.h>
+#include <wrl/client.h>
+
+namespace RED4ext
+{
+namespace GpuApi
+{
+
+enum class CommandListType
+{
+    Invalid,
+    Default,
+    CopySync,
+    CopyAsync,
+    Compute,
+    MAX
+};
+
+struct CommandListContext
+{
+    void AddPendingBarrier(const D3D12_RESOURCE_BARRIER& aBarrier);
+    void Close();
+    void FlushPendingBarriers();
+
+    CString debugName;
+    uint64_t hash;
+    Microsoft::WRL::ComPtr<ID3D12CommandAllocator> commandAllocator;
+    Microsoft::WRL::ComPtr<ID3D12GraphicsCommandList> commandList;
+    uint8_t unk38[0x68 - 0x38];
+    CommandListType type;
+    uint8_t unk6c[0x528 - 0x6c];
+    DynArray<D3D12_RESOURCE_BARRIER> pendingBariers;
+    uint8_t unk538[0x650 - 0x538];
```

Comment:
I would add comments with the offest of the fields, e.g.:

```suggestion
    CString debugName; // 00
    uint64_t hash; // 08
    Microsoft::WRL::ComPtr<ID3D12CommandAllocator> commandAllocator; // 10
    Microsoft::WRL::ComPtr<ID3D12GraphicsCommandList> commandList;
    uint8_t unk38[0x68 - 0x38];
    CommandListType type;
    uint8_t unk6c[0x528 - 0x6c];
    DynArray<D3D12_RESOURCE_BARRIER> pendingBariers;
    uint8_t unk538[0x650 - 0x538];
```

And also some asserts for important stuff, like `pendingBarriers`.